test/mongoose_push_web/controllers/api_v2_notification_controller_test.exs

```diff
@@ -81,43 +67,34 @@ defmodule MongoosePushWeb.APIv2NotificationControllerTest do
     assert json_response(conn, 200) == nil
   end
 
-  test "Request.SendNotification.Deep.SilentNotification schema without required service field",
-       %{
-         conn: conn
-       } do
-    conn =
-      post(
-        conn,
-        "/v2/notification/654321",
-        Jason.encode!(Map.drop(ControllersHelper.silent_request(), ["service"]))
-      )
+  @dropped_fields [{"service", "service"}, {"data", "alert"}]
+  for {dropped, missing} <- @dropped_fields do
+    test "Request.SendNotification.Deep.SilentNotification schema without required #{dropped} field",
+         %{
+           conn: conn
+         } do
+      body =
+        ControllersHelper.silent_request()
+        |> Map.drop([unquote(dropped)])
+        |> Jason.encode!()
 
-    assert json_response(conn, 422) == ControllersHelper.missing_field_response("service")
-  end
+      conn = post(conn, "/v2/notification/654321", body)
 
-  test "Request.SendNotification.Deep.SilentNotification schema without required data field", %{
-    conn: conn
-  } do
-    conn =
-      post(
-        conn,
-        "/v2/notification/654321",
-        Jason.encode!(Map.drop(ControllersHelper.silent_request(), ["data"]))
-      )
-
-    assert json_response(conn, 422) == ControllersHelper.missing_field_response("alert")
+      assert json_response(conn, 422) ==
+               ControllersHelper.missing_field_response(unquote(missing))
+    end
   end
 
   test "Request.SendNotification.Deep.SilentNotification schema with incorrect time_to_live value",
        %{
          conn: conn
        } do
-    conn =
-      post(
-        conn,
-        "/v2/notification/654321",
-        Jason.encode!(%{ControllersHelper.silent_request() | "time_to_live" => "infinity"})
-      )
+    body =
+      ControllersHelper.alert_request()
```

Comment:
Guys, maybe this year' golden shovel award goes to me, but shouldn't it be `silent_request`? 🙄

Reply:
nobody know